config/initializers/active_model_serializers.rb

```diff
@@ -0,0 +1,166 @@
+module AMS
+  module V09
+    class Serializer < ActiveModel::Serializer
+
+       def serializable_hash(adapter_options = nil,
+                             options = {},
+                             adapter_instance = self.class.serialization_adapter_instance)
+         object.nil? ? nil : super
+       end
+    end
+
+    class ArraySerializer < ActiveModel::Serializer::CollectionSerializer
+    end
+
+    class ResourceIdentifier < ActiveModelSerializers::Adapter::JsonApi::ResourceIdentifier
+
+      def initialize(serializer, options)
+        @id   = id_for(serializer)
+        @type = type_for(serializer)
+      end
+
+    private
+
+      def id_for(serializer)
+        serializer.read_attribute_for_serialization(:id)
+      end
+
+    end
+
+    class Base < ActiveModelSerializers::Adapter::Base
+
+      def initialize(serializer, options = {})
+        super
+        @fieldset = options[:fieldset] || ActiveModel::Serializer::Fieldset.new(options.delete(:fields))
+      end
+
+    private
+
+      attr_reader :fieldset
+
+      def attributes_for(serializer, fields)
+        serializer.attributes(fields).except(:id)
+      end
+
+      def resource_object_for(serializer)
+        resource_object = serializer.fetch(self) do
+          resource_object = ResourceIdentifier.new(serializer, instance_options).as_json
+
+          requested_fields = fieldset && fieldset.fields_for(resource_object[:type])
+          attributes = attributes_for(serializer, requested_fields)
+          resource_object.merge!(attributes) if attributes.any?
+          resource_object
+        end
+
+        resource_object
+      end
+
+    end
+
+    # represent resource without associations
+    class Attributes < Base
+
+      def serializable_hash(options = nil)
```

Comment:
- Lint/UnusedMethodArgument: Unused method argument - `options`. If it's necessary, use `_` or `_options` as an argument name to indicate that it won't be used. You can also write as `serializable_hash(*)` if you want the method to accept any arguments but don't care about them.
- Layout/SpaceAroundEqualsInParameterDefault: Surrounding space detected in default value assignment.